src/oas.md

```diff
@@ -557,6 +557,32 @@ servers:
 
 An object representing a Server Variable for server URL template substitution.
 
+The server URL templating is defined by the following [ABNF](https://tools.ietf.org/html/rfc5234) syntax.
+
+```abnf
+server-url-template            = 1*( literals / template-expression )
+template-expression            = "{" template-expression-param-name "}"
+template-expression-param-name = 1*( %x00-79 / %x7C / %x7E-10FFFF ) ; every UTF8 character except { and }
```

Comment:
> > I have kept the naming for template-expression and template-expression-param-name, since they are identical to the ones in path templating. We could imagine "refactoring" them, but it'd probably make things harder. At least implementer know they are the same (should we add a comment in that sense?)

@baywet is it possible to reconsider the naming here? This naming `template-expression` and `template-expression-param-name` comes directly from [openapi-path-templating](https://github.com/swaggerexpert/openapi-path-templating). My intention when creating this naming was to have `template-expression` and `template-expression-param-name` specific to `OpenAPI Path Templating` concept only. The intention is quite explicit from the `template-expression-param-name` non-terminal name specifically - it tries to tells us that the identifier (template variable) inside the brackets represents an OpenAPI `Parameter Object`.`name` fixed field. (which doesn't apply to Server URL templating).

By that logic, that's why I've suggested non-terminals in this ABNF to be server URL templating specific:

```abnf
server-variable        = "{" server-variable-name "}"
server-variable-name   = 1*( %x00-79 / %x7C / %x7E-10FFFF ) ; every UTF8 character except { and }
```

If we want to go for something generic that is identical in both ABNF grammars, I would probably suggest something like:

```abnf
template-expression                = "{" template-expression-identifier "}"
template-expression-identifier = 1*( %x00-79 / %x7C / %x7E-10FFFF ) ; every UTF8 character except { and }
``` 
or
```abnf
template-expression                = "{" template-variable "}"
template-variable = 1*( %x00-79 / %x7C / %x7E-10FFFF ) ; every UTF8 character except { and }
``` 

But IMHO there is benefit in keeping the naming distinct as it will allow to evolve those non-terminals independently in future. I can imagine, future OpenAPI versions might adopt further concepts (expadning) from RFC 6570, while it doesn't make sense for Server URL templating to adopt those.

Reply:
I can buy the future changes that diverge, creating confusion. Updated to a distinct name, let me know if I missed anything.